super_dpf.py

```diff
@@ -1,127 +1,316 @@
+        user_id = raw_input('Google Photos Username: ')
+        album_id = raw_input('Album ID: ')
+        enable = raw_input('Enter "yes" to enable Google Photo sync: ')
+
+        enable = 'true' if enable.lower() == 'yes' else ''
+
+        if not os.path.isdir(self.gphotos_dir):
+            os.makedirs(self.gphotos_dir)
+
+        self._save_settings(g_enabled=enable, feed_url=list(feed_url),
+                            user_id=user_id, album_id=album_id)
+
+    def _gphotos_sync(self):
+        if not self.gphotos_configured:
+            logging.info('Google Photos is not configured.')
+            configure = raw_input('Configure it now? ')
+            if configure.lower() == 'yes':
+                self.gphotos_configure_dialogue()
+        elif not self.gphotos_enabled:
+            logging.info('Google Photos is not enabled.')
+            enable = raw_input('Enable it now? ')
+            enable = 'true' if enable.lower() == 'yes' else ''
+            self._save_settings(g_enabled=enable)
+        else:
+            feed_url = self.settings['gphotos'].get('feed_url')
+            for feed in feed_url:
+                response = requests.get(feed)
+                tags = BeautifulStoneSoup(
+                    response.content).findAll('media:content')
+                for tag in tags:
+                    if tag.get('url', False):
+                        filename = os.path.join(
+                            self.gphotos_dir,
+                            os.path.basename(urlsplit(tag['url'])[2]))
+                        if not os.path.isfile(filename):
+                            logging.info(
+                                'Getting img at {}'.format(tag['url']))
+                            response = requests.get(tag['url'], stream=True)
+                            try:
+                                img = Image.open(StringIO(response.content))
+                                img.save('{}'.format(filename), optimize=True,
+                                         progressive=True, quality=100,
+                                         subsampling=0)
+                            except IOError as e:
+                                logging.error(e)
+
+
+class DropboxController(BaseDPF):
+    pass
+
+
+class FlickrController(BaseDPF):
+    # https://stuvel.eu/media/flickrapi-docs/documentation/
+    pass
+
+
+class InstagramController(BaseDPF):
+    # https://github.com/Instagram/python-instagram
+    pass
+
+
+class HotMediaController(BaseDPF):
+    pass
+
+
+class SuperDPF(AWSController, GPhotoController):
+    def _sync(self):
+        if self.aws_enabled:
+            logging.info('Syncing AWS....')
+            self._aws_sync()
+
+        if self.gphotos_enabled:
+            logging.info('Syncing Google Photos....')
+            self._gphotos_sync()
+
+    def run(self):
+        self._sync()
+        os.system('/usr/bin/supervisorctl restart sdpf')
```

Comment:
supervisorctl should be assumed to be in the path.  It may however be prudent to make sure that you are either running against the system level instance of supervisord or make sure you are in the right directory to act on the user level instance that you have previously started.